File: languages/R/src/provider/MinValueAnalysis.R
# Criando uma classe "MinValueAnalysis"
MinValueAnalysis = 
  R6Class("MinValueAnalysis",
          # Definindo valores e funções públicas
          public = list(
            
            # Método de interface, responsável por realizar a análise na lista de "UserInfo"
            # list -> Lista de "UserInfo"
            # RETURN -> Valor minimo da lista            
            analysis = function(userInfoList) {
              
              creditVector = sapply(userInfoList, function(userInfo) {
                return(userInfo$getCredit())
              })
              
              return(min(creditVector))
            }
          )
  )

File: languages/java/src/provider/MinValueAnalysis.java
package provider;

import model.SimpleTableAnalysis;
import model.UserInfo;

import java.util.List;

public class MinValueAnalysis implements SimpleTableAnalysis {

    @Override
    public double analysis(List<UserInfo> userInfoList) {
        double min = Double.MAX_VALUE;
        for(UserInfo userInfo : userInfoList){
            double credit = userInfo.getCredit();
            if(min > credit) min = credit;
        }
        return min;
    }
}

